src/data/finite/set.lean

```diff
@@ -1,168 +0,0 @@
-  haveI := fintype.of_finite (plift ι),
-  haveI := λ i, fintype.of_finite (f i),
-  apply_instance,
-end
-
-instance finite_sUnion {s : set (set α)} [finite s] [H : ∀ (t : s), finite (t : set α)] :
-  finite (⋃₀ s) :=
-by { rw sUnion_eq_Union, exact @finite.set.finite_Union _ _ _ _ H }
-
-lemma finite_bUnion {ι : Type*} (s : set ι) [finite s] (t : ι → set α) (H : ∀ i ∈ s, finite (t i)) :
-  finite (⋃(x ∈ s), t x) :=
-begin
-  convert_to finite (⋃ (x : s), t x),
-  { congr' 1, ext, simp },
-  haveI : ∀ (i : s), finite (t i) := λ i, H i i.property,
-  apply_instance,
-end
-
-instance finite_bUnion' {ι : Type*} (s : set ι) [finite s] (t : ι → set α) [∀ i, finite (t i)] :
-  finite (⋃(x ∈ s), t x) :=
-finite_bUnion s t (λ i h, infer_instance)
-
-/--
-Example: `finite (⋃ (i < n), f i)` where `f : ℕ → set α` and `[∀ i, finite (f i)]`
-(when given instances from `data.nat.interval`).
--/
-instance finite_bUnion'' {ι : Type*} (p : ι → Prop) [h : finite {x | p x}]
-  (t : ι → set α) [∀ i, finite (t i)] :
-  finite (⋃ x (h : p x), t x) :=
-@finite.set.finite_bUnion' _ _ (set_of p) h t _
-
-instance finite_Inter {ι : Sort*} [nonempty ι] (t : ι → set α) [∀ i, finite (t i)] :
-  finite (⋂ i, t i) :=
-finite.set.subset (t $ classical.arbitrary ι) (Inter_subset _ _)
-
-instance finite_insert (a : α) (s : set α) [finite s] : finite (insert a s : set α) :=
-((set.finite_of_finite s).insert a).finite
-
-instance finite_image (s : set α) (f : α → β) [finite s] : finite (f '' s) :=
-((set.finite_of_finite s).image f).finite
-
-instance finite_range (f : ι → α) [finite ι] : finite (range f) :=
-by { haveI := fintype.of_finite (plift ι), apply_instance }
-
-instance finite_replacement [finite α] (f : α → β) : finite {(f x) | (x : α)} :=
-finite.set.finite_range f
-
-instance finite_prod (s : set α) (t : set β) [finite s] [finite t] :
-  finite (s ×ˢ t : set (α × β)) :=
-by { haveI := fintype.of_finite s, haveI := fintype.of_finite t, apply_instance }
-
-instance finite_image2 (f : α → β → γ) (s : set α) (t : set β) [finite s] [finite t] :
-  finite (image2 f s t : set γ) :=
-by { rw ← image_prod, apply_instance }
-
-instance finite_seq (f : set (α → β)) (s : set α) [finite f] [finite s] : finite (f.seq s) :=
-by { rw seq_def, apply_instance }
-
-end finite.set
-
-/-! ### Non-instances -/
-
-lemma set.finite_univ_iff : finite (set.univ : set α) ↔ finite α :=
-(equiv.set.univ α).finite_iff
-
-lemma finite.of_finite_univ [finite ↥(univ : set α)] : finite α :=
-set.finite_univ_iff.mp ‹_›
-
-lemma finite.set.finite_of_finite_image (s : set α)
-  {f : α → β} (h : s.inj_on f) [finite (f '' s)] : finite s :=
-finite.of_equiv _ (equiv.of_bijective _ h.bij_on_image.bijective).symm
-
-lemma finite.of_injective_finite_range {f : α → β}
-  (hf : function.injective f) [finite (range f)] : finite α :=
-begin
-  refine finite.of_injective (set.range_factorization f) (λ x y h, hf _),
-  simpa only [range_factorization_coe] using congr_arg (coe : range f → β) h,
-end
```

Comment:
They were added after I created my branch. I'm not sure that we should duplicate lots of API about finite sets with `[finite s]` assumptions. I'm adding `set.finite` versions of these lemmas. In particular, `function.injective.finite_range_iff` works for a function from a `Sort*`.

Reply:
The first three had been there since the beginning, but `finite.of_injective_finite_range` is new. The `finite.set.finite_of_finite_image` lemma has simpler hypotheses than the `fintype` version since it doesn't need to try to be computable.

I haven't thought too much about whether `finite.of_injective_finite_range` should exist, but I'm not comfortable deleting a lemma that had just been merged four days ago... At least we can delete it in a simpler PR.